livekit-agents/livekit/agents/utils/av_sync.py

```diff
@@ -0,0 +1,173 @@
+import asyncio
+import logging
+import time
+from collections import deque
+from typing import Optional, Union
+
+import livekit.agents.utils as utils
+from livekit import rtc
+
+logger = logging.getLogger(__name__)
+
+
+class AVSynchronizer:
+    """Synchronize audio and video capture.
+
+    Usage:
+        av_sync = AVSynchronizer(
+            audio_source=audio_source,
+            video_source=video_source,
+            video_sample_rate=video_sample_rate,
+        )
+
+        async for video_frame, audio_frame in video_generator:
+            av_sync.push(video_frame)
```

Comment:
```suggestion
            await av_sync.push(video_frame)
```

Reply:
for docs, maybe also clarify that it could be pushed from different tasks?

Reply:
Yeah, I'll create an example to test and demonstrate the use case.

Reply:
this example is awesome! it'll be super useful. It might be useful to include an example file too.. I'll see about moving some of these examples to its own organized repo in livekit-examples org

Reply:
Yes, or I was thinking of another example of reading video and audio from the client and sending it back, or an audio wave with a graph so we can easily see the synchronization while not needing an additional mp4 file.